### PR TITLE
Nested block comments

### PR DESCRIPTION
**Pull request by susliko**
Fixes https://github.com/tree-sitter/tree-sitter-scala/issues/226

Problem
-------
Nested block comments are not parsed causing an error, e.g.
```
/* /* */ */
```

Solution
-------
Leave `$.comment` for single-line comments.
Introduce `$.block_comment` for block-comments.
Make `$.block_comment` out of several tokens to be able to detect nested block comments

In the future we'll probably want to add support for scala-cli directives inside `$.comment`and scaladoc annotations inside `$.block_comment`  to provide syntax highlighting

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1309,8 +1310,8 @@ module.exports = grammar({
       seq(
         // opchar
         /[\-!#%&*+\/\\:<=>?@\u005e\u007c~\p{Sm}\p{So}]/,
-        // opchar minus slash
-        /[\-!#%&*+\\:<=>?@\u005e\u007c~\p{Sm}\p{So}]/,
+        // opchar minus slash and asterisk
+        /[\-!#%&+\\:<=>?@\u005e\u007c~\p{Sm}\p{So}]/,
```
Comment (eed3si9n) [inline grammar.js:1314]: Would this cause issue matching on `**`?

Comment (susliko) [reply grammar.js:1314]: Somehow it doesn't: https://github.com/tree-sitter/tree-sitter-scala/pull/251/commits/8f1f347918b6acce1b0718d79c26f2ae9dfb92b9

Comment (susliko) [reply grammar.js:1314]: But this change is essential for this PR
Without it the comment start in the following code is parsed as operator_identifier
```scala
object X:
  /** foo */
```

Comment (eed3si9n) [reply grammar.js:1314]: I guess `**` would match `repeat(/[\-!#%&*+\/\\:<=>?@\u005e\u007c~\p{Sm}\p{So}]/),` of the `choice` branch.

**Review by eed3si9n — APPROVED** [review]
Thanks!

**Pull request merged**